api/api-internal-environments-v1/src/main/java/com/thoughtworks/go/apiv1/internalenvironments/InternalEnvironmentsControllerV1.java

```diff
@@ -66,6 +75,7 @@ public void setupRoutes() {
 
             get("", mimeType, this::index);
             get("/merged", mimeType, this::indexMergedEnvironments);
+            put(Routes.InternalEnvironments.ENV_NAME, mimeType, this::updateAgentAssociation);
```

Comment:
I am not sure if we use internal APIs for modifications. We use them just to get data in the format required by view. Can you please confirm once?

Reply:
As discussed with @maheshp, we have decided to keep this as it is as this is a very customized request for our front end.